src/components/Header/Header.js

```diff
@@ -1,7 +1,107 @@
 import React from "react";
+import { Link } from "react-router-dom";
+import styled from "styled-components";
 
 const Header = () => {
-  return <div>헤더</div>;
+  const token = true;
+
+  return (
+    <HeaderFlex>
+      <Link to="/">LOGO</Link>
+      <CategoryFlex>
+        <CategoryName>
+          <Link to="/">예약하기</Link>
+        </CategoryName>
+        <CategoryName>
+          <Link to="Matching">매칭하기</Link>
+        </CategoryName>
+      </CategoryFlex>
+      <CategoryFlex>
+        {!token ? (
+          <LoginButton>로그인 및 회원가입</LoginButton>
+        ) : (
+          <LoginButton>로그아웃</LoginButton>
+        )}
```

Comment:
```suggestion
          <LoginButton>{!token ? "로그인 및 회원가입" : "로그아웃"}</LoginButton>
```
이렇게 구현하는게 조금 더 깔끔하겠네요!